src/SSHLibrary/pythonclient.py

```diff
@@ -129,8 +132,8 @@ def _list(self, path):
         path = path.encode(self._encoding)
         for item in self._client.listdir_attr(path):
             filename = item.filename
-            if not isinstance(filename, unicode):
-                filename = unicode(filename, self._encoding)
+            if not is_string(filename):
+                filename = unic(filename)
```

Comment:
- Decoding to Unicode should be done only if `filename` is Unicode and should thus use `is_unicode`, not `is_string`. The latter returns true also with byte strings on Python 2.
- Instead of blindly converting to Unicode with `unic`, should use `filename.decode(self._encoding)`.
- I'm surprised the code needs to check is `filename` Unicode or not. Doesn't Paramiko always return it in same format?

Reply:
ok